### PR TITLE
chore: add base code for NestJS along with linting and CI

### PR DESCRIPTION
**Pull request by nitesh-osmosys**
### Pre-requisites

- [x] I have gone through the Contributing guidelines for [Submitting a Pull Request (PR)](../../CONTRIBUTING.md#submitting-a-pull-request-pr) and ensured that this is not a duplicate PR.
- [x] I have performed preliminary unit testing.
- [ ] I have updated the required [api docs](../../apps/api/docs/) as applicable.
- [ ] I have added/updated test cases to the [test suite](../../apps/api/TranscriptSummarizer.postman_collection.json) as applicable

### PR Details

PR details have been updated as per the given format (see below)

- [x] PR title adheres to the format specified in guidelines (e.g., `feat: add admin login endpoint`)
- [x] Description has been added
- [ ] Related changes have been added (optional)
- [ ] Screenshots have been added (optional)
- [ ] Query request and response examples have been added (as applicable, in case added or updated)
- [ ] Documentation changes have been listed (as applicable)
- [ ] Test suite output is added (as applicable)
- [ ] Pending actions have been added (optional)
- [ ] Any other additional notes have been added (optional)

### Additional Information

- [x] Appropriate label(s) have been added (`ready for review` should be added if the PR is ready to be reviewed)
- [ ] Assignee(s) and reviewer(s) have been added (optional)

---

**Description:**

- Added code base for nest js along with linting rules and basic package libraries

**Related changes:**

- Added config->logger.config.ts file
- Updated main.ts
- Added linting rules
- Installed required packages

**Screenshots:**

image

**Query request and response:**

- Add any query request body, URL statement and response body for the made change or addition.

**Documentation changes:**

- NA

**Test suite output:**

- NA

**Pending actions:**

- NA

**Additional notes:**

- NA

## Summary by CodeRabbit

- **New Features**
  - Added a greeting message feature to the root route in the API.
  - Introduced a GitHub Actions workflow for linting and building the Nest.js application on pull requests to the main branch.
- **Documentation**
  - Introduced comprehensive documentation for the API including installation, running, and testing instructions.
- **Testing**
  - Implemented unit tests for the `AppController` and end-to-end tests for the API.
- **Chores**
  - Configured ESLint, Prettier, and Jest for enhanced code quality and testing.
- **Bug Fixes**
  - Improved logging with file rotation to enhance diagnostics and troubleshooting.

### CODE REVIEW

File: apps/api/package.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
+{
+  "name": "api",
+  "version": "0.0.1",
+  "description": "",
+  "author": "",
+  "private": true,
+  "license": "UNLICENSED",
```
Comment (xixas) [inline apps/api/package.json:7]: Why is this unlicensed? It should be MIT. Did you add this value manually?

File: apps/api/package.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  "scripts": {
```
Comment (xixas) [inline apps/api/package.json:8]: Add node version as 20
Check this
https://github.com/OsmosysSoftware/osmo-x/blob/399d75882f54c9e1dbb2f472d7c5d86feaa01227/apps/api/package.json#L7

**Review by xixas — CHANGES_REQUESTED** [review]

File: apps/api/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    "outDir": "./dist",
+    "baseUrl": "./",
+    "incremental": true,
+    "skipLibCheck": true,
+    "strictNullChecks": false,
+    "noImplicitAny": false,
+    "strictBindCallApply": false,
+    "forceConsistentCasingInFileNames": false,
+    "noFallthroughCasesInSwitch": false
+  }
```
[the remark this answers is not in the document]
Comment (xixas) [reply apps/api/tsconfig.json:20]: Can you confirm if you took the tsconfig from osmox?

File: apps/api/package.json (hunk 1 above)
Comment (nitesh-osmosys) [reply apps/api/package.json:7]: No, it is a default value only, I didn't make any changes in this

File: apps/api/package.json (hunk 2 above)
Comment (nitesh-osmosys) [reply apps/api/package.json:8]: Updated

**Review by xixas — APPROVED** [review]

**Comment by xixas** [comment]
@nitesh-osmosys 
Add this CI file in repo
https://github.com/OsmosysSoftware/osmo-x/blob/main/.github/workflows/backend.yml

**Pull request merged**